src/modules/commander/Commander.cpp

```diff
@@ -1395,8 +1396,13 @@ Commander::run()
 
 	int32_t arm_mission_required_param = 0;
 	param_get(_param_arm_mission_required, &arm_mission_required_param);
-	arm_requirements |= (arm_mission_required_param &
-			     (PreFlightCheck::ARM_REQ_MISSION_BIT | PreFlightCheck::ARM_REQ_ARM_AUTH_BIT));
+	arm_requirements |= (arm_mission_required_param == 0) ? PreFlightCheck::ARM_REQ_NONE :
+			    PreFlightCheck::ARM_REQ_MISSION_BIT;
```

Comment:
It's correct but actually a bit weird to bitwise or with 0.

What do you think about?
```
if (arm_mission_required_param == 1) {
    arm_requirements |= PreFlightCheck::ARM_REQ_MISSION_BIT;
}

I can see that it's done like that everywhere below so it would be inconsistent, so in that case leave it or change it all.

Reply:
I Agree. 
I'd rather keep it as it is for this PR and then come up with an incremental PR where the logic is refactored and also code duplication is removed.